src/libcore/mem.rs

```diff
@@ -611,3 +611,38 @@ pub const POST_DROP_USIZE: usize = POST_DROP_U64 as usize;
 pub unsafe fn transmute_copy<T, U>(src: &T) -> U {
     ptr::read(src as *const T as *const U)
 }
+
+/// Returns the value of the discriminant for the enum variant in `v`.
+///
+/// The returned discriminant is cast to `u64`, no matter what the actual
```

Comment:
>  If a discriminant isn't specified, they start at zero, and add one for each variant, in order.

This is for C-like (i.e. numeric/data-less) enums only. We do not specify anything for sum-types. I would like to not guarantee anything other than `discriminant(Enum::Variant) == discriminant(Enum::Variant)` except for C-like enums, where the usual increment-by-one rules apply.